api.go

```diff
@@ -645,6 +645,14 @@ func (r *Raft) restoreSnapshot() error {
 		r.setCommittedConfiguration(conf, index)
 		r.setLatestConfiguration(conf, index)
 
+		// Remove old logs if r.logs is a MonotonicLogStore. Log any errors and
+		// continue.
+		if logs, ok := r.logs.(MonotonicLogStore); ok && logs.IsMonotonic() {
+			if err := r.removeOldLogs(); err != nil {
+				r.logger.Error("failed to reset logs", "error", err)
+			}
+		}
+
```

Comment:
This change is not necessary and in fact will _loose data_ in raft.

This method is called on startup to restore from the snapshot on disk which is very different from restoring a user snapshot (which invalidates our current state).

As written, this would delete all the recently written things from disk on startup - if we happen to restart or crash again before the next snapshot we would _loose data_.

So we should remove this from here!

Reply:
Paired with @banks today to get context.  The correction to the test and the code is contained in this commit: https://github.com/hashicorp/raft/pull/545/commits/dd083107e93e2ab7c7b0679d38bd225e0fc6720c